Generate serialization for WebCore::Payment, WebCore::PaymentContact, and WebCore::PaymentMerchantSession

sheeparegreat opened the pull request:
#### 2a16c10dc012b0a6e2f7c69ada165274fec5bed8
<pre>
Generate serialization for WebCore::Payment, WebCore::PaymentContact, and WebCore::PaymentMerchantSession
<a href="https://bugs.webkit.org/show_bug.cgi?id=267703">https://bugs.webkit.org/show_bug.cgi?id=267703</a>
<a href="https://rdar.apple.com/121195605">rdar://121195605</a>

Reviewed by Alex Christensen.

* Source/WebCore/Modules/applepay/Payment.h:
* Source/WebCore/Modules/applepay/PaymentContact.h:
* Source/WebCore/Modules/applepay/PaymentMerchantSession.h:
(WebCore::PaymentMerchantSession::pkPaymentMerchantSession const):
* Source/WebCore/Modules/applepay/cocoa/PaymentCocoa.mm:
(WebCore::Payment::pkPayment const):
* Source/WebCore/Modules/applepay/cocoa/PaymentContactCocoa.mm:
(WebCore::PaymentContact::pkContact const):
* Source/WebCore/testing/MockPaymentCoordinator.cpp:
(WebCore::MockPaymentCoordinator::showPaymentUI):
* Source/WebKit/Platform/cocoa/PaymentAuthorizationPresenter.mm:
(WebKit::PaymentAuthorizationPresenter::completeMerchantValidation):
* Source/WebKit/Shared/ApplePay/cocoa/WebPaymentCoordinatorProxyCocoa.mm:
(WebKit::WebPaymentCoordinatorProxy::platformPaymentRequest):
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.mm:
(IPC::getClass&lt;PKPayment&gt;): Deleted.
(IPC::getClass&lt;PKPaymentMerchantSession&gt;): Deleted.
(IPC::getClass&lt;PKPaymentMethod&gt;): Deleted.
(IPC::ArgumentCoder&lt;WebCore::Payment&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;WebCore::Payment&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;WebCore::PaymentContact&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;WebCore::PaymentContact&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;WebCore::PaymentMerchantSession&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;WebCore::PaymentMerchantSession&gt;::decode): Deleted.
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.h:

Canonical link: <a href="https://commits.webkit.org/273512@main">https://commits.webkit.org/273512@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4dc79c545b89d280aaa9d07e7a2788b340bbdee4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35566 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14506 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/37702 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38306 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32050 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/36764 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/16932 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11534 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/30862 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36119 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12292 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31657 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/10764 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/10808 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/31790 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39551 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32355 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32126 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/36741 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/10963 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/8854 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/34800 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/12678 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11477 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4617 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/11743 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (commented):
LGTM

achristensen07 commented on Source/WebCore/Modules/applepay/Payment.h:
```diff
@@ -46,6 +47,7 @@ class WEBCORE_EXPORT Payment {
     PKPayment *pkPayment() const;
 
 private:
+    friend struct IPC::ArgumentCoder<Payment, void>;
```
Let's instead make pkPayment() return a RetainPtr.  Same with the others.

achristensen07 commented on Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in:
```diff
@@ -113,6 +113,18 @@ headers: <pal/cocoa/PassKitSoftLink.h>
 }
+
+class WebCore::Payment {
+    [Precondition='PAL::isPassKitCoreFrameworkAvailable()', SecureCodingAllowed=[PAL::getPKPaymentClass()], Validator='pkPayment'] RetainPtr<PKPayment> pkPayment();
```
The validators are not equivalent to what was there before.  decoder.decodeWithAllowedClasses<T> returns an optional<RetainPtr<T>>.  Same with WebCore::PaymentMethod whose validator should be removed.  Null is ok, decode failure is not.

sheeparegreat committed:
Generate serialization for WebCore::Payment, WebCore::PaymentContact, and WebCore::PaymentMerchantSession
https://bugs.webkit.org/show_bug.cgi?id=267703
rdar://121195605

Reviewed by Alex Christensen.

* Source/WebCore/Modules/applepay/Payment.h:
* Source/WebCore/Modules/applepay/PaymentContact.h:
* Source/WebCore/Modules/applepay/PaymentMerchantSession.h:
(WebCore::PaymentMerchantSession::pkPaymentMerchantSession const):
* Source/WebCore/Modules/applepay/cocoa/PaymentCocoa.mm:
(WebCore::Payment::pkPayment const):
* Source/WebCore/Modules/applepay/cocoa/PaymentContactCocoa.mm:
(WebCore::PaymentContact::pkContact const):
* Source/WebCore/testing/MockPaymentCoordinator.cpp:
(WebCore::MockPaymentCoordinator::showPaymentUI):
* Source/WebKit/Platform/cocoa/PaymentAuthorizationPresenter.mm:
(WebKit::PaymentAuthorizationPresenter::completeMerchantValidation):
* Source/WebKit/Shared/ApplePay/cocoa/WebPaymentCoordinatorProxyCocoa.mm:
(WebKit::WebPaymentCoordinatorProxy::platformPaymentRequest):
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.mm:
(IPC::getClass<PKPayment>): Deleted.
(IPC::getClass<PKPaymentMerchantSession>): Deleted.
(IPC::getClass<PKPaymentMethod>): Deleted.
(IPC::ArgumentCoder<WebCore::Payment>::encode): Deleted.
(IPC::ArgumentCoder<WebCore::Payment>::decode): Deleted.
(IPC::ArgumentCoder<WebCore::PaymentContact>::encode): Deleted.
(IPC::ArgumentCoder<WebCore::PaymentContact>::decode): Deleted.
(IPC::ArgumentCoder<WebCore::PaymentMerchantSession>::encode): Deleted.
(IPC::ArgumentCoder<WebCore::PaymentMerchantSession>::decode): Deleted.
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.h:

Canonical link: https://commits.webkit.org/273512@main
